tools, toc: reorganize TiDB Binlog documents (#1078)
pingcap/docs: tools/binlog/deploy.md

---
title: TiDB-Binlog deployment

> summary: Learn how to deploy the cluster version of TiDB-Binlog.

Reviewer: ```suggestion
summary: Learn how to deploy TiDB-Binlog cluster.
```

        ```ini
        ## binlog trigger
        enable_binlog = True
        ```

    2. Add the deployment machine IPs for `pump_servers`.

        ```ini
        ## Binlog Part
        [pump_servers]
        172.16.10.72
        172.16.10.73
        172.16.10.74
        ```

        Pump retains the data of the latest 7 days by default. You can modify the value of the `gc` variable in the `tidb-ansible/conf/pump.yml` file and remove the related comments: 
        
        ```yaml
        global:
          # an integer value to control the expiry date of the binlog data, which indicates for how long (in days) the binlog data would be stored
          # must be bigger than 0
          # gc: 7
        ```

        Make sure the space of the deployment directory is sufficient for storing Binlog. For more details, see [Configure the deployment directory](../../dev/how-to/deploy/orchestrated/ansible.md#configure-the-deployment-directory). You can also set a separate deployment directory for Pump.

        ```ini
        ## Binlog Part
        [pump_servers]
        pump1 ansible_host=172.16.10.72 deploy_dir=/data1/pump
        pump2 ansible_host=172.16.10.73 deploy_dir=/data2/pump

>         pump3 ansible_host=172.16.10.74 deploy_dir=/data1/pump

Reviewer: ```suggestion
        pump3 ansible_host=172.16.10.74 deploy_dir=/data3/pump
```

    2018/06/21 11:24:47 meta.go:117: [info] meta: &{CommitTS:400962745252184065}
    ```

    This command outputs `meta: &{CommitTS:400962745252184065}`, and the value of `CommitTS` is used as the value of the `initial-commit-ts` parameter needed for the initial start of Drainer.

2. Modify the `tidb-ansible/inventory.ini` file.

    Add the deployment machine IPs for `drainer_servers`. Set `initial_commit_ts` to the value you have obtained, which is only used for the initial start of Drainer.

    - Assume that the downstream is MySQL with the alias `drainer_mysql`:

        ```ini
        [drainer_servers]
        drainer_mysql ansible_host=172.16.10.71 initial_commit_ts="402899541671542785"
        ```

    - Assume that the downstream is `file` with the alias `drainer_file`:

        ```ini
        [drainer_servers]
        drainer_file ansible_host=172.16.10.71 initial_commit_ts="402899541671542785"
        ```

3. Modify the configuration file.

    - Assume that the downstream is MySQL:

        ```bash
        $ cd /home/tidb/tidb-ansible/conf
        $ cp drainer.toml drainer_mysql_drainer.toml
        $ vi drainer_mysql_drainer.toml
        ```

>         > **Note:** Name the configuration file as `alias_drainer.toml`. Otherwise, the customized configuration file cannot be found during the deployment process.

Reviewer 2: ```suggestion
        > **Note:**
        >
        > Name the configuration file as `alias_drainer.toml`. Otherwise, the customized configuration file cannot be found during the deployment process.
```

        # db-name ="test"
        # tbl-name = "log"

        # [[syncer.replicate-do-table]]
        # db-name ="test"
        # tbl-name = "~^a.*"

        # Ignore the replication of some tables
        # [[syncer.ignore-table]]
        # db-name = "test"
        # tbl-name = "log"

        # the server parameters of the downstream database when `db-type` is set to "mysql"
        [syncer.to]
        host = "192.168.0.13"
        user = "root"
        password = ""
        port = 3306

        # the directory where the binlog file is stored when `db-type` is set to `file`
        # [syncer.to]
        # dir = "data.drainer"

        # the Kafka configuration when `db-type` is set to "kafka"
        # [syncer.to]
        # zookeeper-addrs = "127.0.0.1:2181"
        # kafka-addrs = "127.0.0.1:9092"
        # kafka-version = "0.8.2.0"

        # the topic name of the Kafka cluster that saves the binlog data. The default value is <cluster-id>_obinlog
        # To run multiple Drainers to replicate data to the same Kafka cluster, you need to set different `topic-name`s for each Drainer.
        # topic-name = ""
        ```

    - Starting Drainer:

>         > **Note:** If the downstream is MySQL/TiDB, to guarantee the data integrity, you need to obtain the `initial-commit-ts` value and make a full backup of the data and restore the data before the initial start of Drainer. For details, see [Deploy Drainer](#step-3-deploy-drainer).

Reviewer 2: ```suggestion
        > **Note:**
        >
        > If the downstream is MySQL/TiDB, to guarantee the data integrity, you need to obtain the `initial-commit-ts` value and make a full backup of the data and restore the data before the initial start of Drainer. For details, see [Deploy Drainer](#step-3-deploy-drainer).
```

---
title: TiDB-Binlog Cluster deployment
summary: Learn how to deploy TiDB-Binlog cluster.
category: tools
---

# TiDB-Binlog Cluster deployment

> This section shows two methods of deploying TiDB-Binlog:

Reviewer 2: ```suggestion
This document describes two methods of deploying TiDB-Binlog:
```

        ```

    - Download the 2.1 version:

        ```bash
        $ git clone -b release-2.1 https://github.com/pingcap/tidb-ansible.git
        ```

    - Download the master version:

        ```bash
        $ git clone https://github.com/pingcap/tidb-ansible.git
        ```
        
### Step 2: Deploy Pump
1. Modify the `tidb-ansible/inventory.ini` file.

    1. Set `enable_binlog = True` to start `binlog` of the TiDB cluster.

        ```ini
        ## binlog trigger
        enable_binlog = True
        ```

    2. Add the deployment machine IPs for `pump_servers`.

        ```ini
        ## Binlog Part
        [pump_servers]
        172.16.10.72
        172.16.10.73
        172.16.10.74
        ```

        Pump retains the data of the latest 7 days by default. You can modify the value of the `gc` variable in the `tidb-ansible/conf/pump.yml` file and remove the related comments: 
        
        ```yaml
        global:
          # an integer value to control the expiry date of the binlog data, which indicates for how long (in days) the binlog data would be stored
          # must be bigger than 0
          # gc: 7
        ```

>         Make sure the space of the deployment directory is sufficient for storing Binlog. For more details, see [Configure the deployment directory](../../dev/how-to/deploy/orchestrated/ansible.md#configure-the-deployment-directory). You can also set a separate deployment directory for Pump.

Reviewer 2: Let's use the following link path rule because we are refactoring documentation these days. This rule can avoid possible issues.

```suggestion
        Make sure the space of the deployment directory is sufficient for storing Binlog. For more details, see [Configure the deployment directory](/dev/how-to/deploy/orchestrated/ansible.md#configure-the-deployment-directory). You can also set a separate deployment directory for Pump.
```

        ```ini
        ## Binlog Part
        [pump_servers]
        pump1 ansible_host=172.16.10.72 deploy_dir=/data1/pump
        pump2 ansible_host=172.16.10.73 deploy_dir=/data2/pump
        pump3 ansible_host=172.16.10.74 deploy_dir=/data3/pump
        ```

2. Deploy and start the TiDB cluster containing Pump.

    After configuring the `inventory.ini` file, you can choose one method from below to deploy the TiDB cluster.

    **Method #1**: Add Pump on the existing TiDB cluster.

    1. Deploy `pump_servers` and `node_exporters`.

        ```
        ansible-playbook deploy.yml -l ${pump1_ip}, ${pump2_ip}, [${alias1_name}, ${alias2_name}]
        ```

    2. Start `pump_servers`.

        ```
        ansible-playbook start.yml --tags=pump
        ```

    3. Update and restart `tidb_servers`.

        ```
        ansible-playbook rolling_update.yml --tags=tidb
        ```

    4. Update the monitoring data.

        ```
        ansible-playbook rolling_update_monitor.yml --tags=prometheus
        ```

    **Method #2**: Deploy a TiDB cluster containing Pump from scratch.

>     For how to use Ansible to deploy the TiDB cluster, see [Deploy TiDB Using Ansible](../../dev/how-to/deploy/orchestrated/ansible.md).

Reviewer 2: ```suggestion
    For how to use Ansible to deploy the TiDB cluster, see [Deploy TiDB Using Ansible](/dev/how-to/deploy/orchestrated/ansible.md).
```

---

> title: TiDB-Binlog Cluster deployment

Reviewer 2: ```suggestion
title: TiDB-Binlog Cluster Deployment
```

summary: Learn how to deploy TiDB-Binlog cluster.
category: tools
---

> # TiDB-Binlog Cluster deployment

Reviewer 2: ```suggestion
# TiDB-Binlog Cluster Deployment
```

This document describes two methods of deploying TiDB-Binlog:

- [Deploy TiDB-Binlog using TiDB-Ansible](#deploy-tidb-binlog-using-tidb-ansible)

> - [Deploy TiDB-Binlog using a Binary package](#deploy-tidb-binlog-using-binary)

Reviewer 2: ```suggestion
- [Deploy TiDB-Binlog using a Binary package](#deploy-tidb-binlog-using-a-binary-package)
```
